sharktank/tests/models/llama/benchmark_tests.py

```diff
@@ -0,0 +1,837 @@
+        """
+        cmd = self.get_compile_cmd(
+            output_mlir_path=mlir_path,
+            output_vmfb_path=output_vmfb_path,
+            args=args,
+        )
+        logging.getLogger().info(f"Launching compile command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, capture_output=True, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to compile.")
+
+    def iree_benchmark_module(
+        self,
+        *,
+        hip_device_id: str,
+        vmfb_name: str,
+        irpa_path: str,
+        args: List[str],
+        cwd: str | Path,
+    ):
+        """Runs a compiled program with the given args using `iree-benchmark-module`.
+        This assumes that the `iree-benchmark-module` command is available (usually via PATH).
+        Args:
+            vmfb_name: Name of the .vmfb file (relative to `cwd`).
+            args: List of arguments to pass to `iree-benchmark-module`.
+            cwd: Working directory to run the command within. (either string or Path works)
+            compile_cmd: Command used to compile the program, for inclusion in error messages.
+        Raises Exception if running fails for some reason.
+        """
+        benchmark_args = [
+            f"ROCR_VISIBLE_DEVICES={hip_device_id}",
+            "iree-benchmark-module",
+            f"--device=hip://{hip_device_id}",
+            "--hip_use_streams=true",
+            "--hip_allow_inline_execution=true",
+            "--device_allocator=caching",
+            f"--module={vmfb_name}",
+            f"--parameters=model={irpa_path}",
+        ]
+        benchmark_args += args
+        cmd = subprocess.list2cmdline(benchmark_args)
+        logging.getLogger().info(f"Launching run command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, stdout=sys.stdout, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to run")
+
+
+class BenchmarkLlama3_1_8B(BaseBenchmarkTest):
+    def setUp(self):
+        super().setUp()
+        # TODO: add numpy files to Azure and download from it
+        artifacts_dir = "/data/extra/models/llama3.1_8B/"
+        self.irpa_path = artifacts_dir + "llama8b_f16.irpa"
+        self.irpa_path_fp8 = artifacts_dir + "llama8b_fp8.irpa"
+        self.tensor_parallelism_size = None
+        self.dir_path_8b = self.dir_path + "/llama-8b/"
+        self.temp_dir_8b = Path(self.dir_path_8b)
+        self.temp_dir_8b.mkdir(parents=True, exist_ok=True)
+        self.iree_compile_args = [
+            "--iree-hal-target-backends=rocm",
+            "--iree-hip-target=gfx942",
+        ]
+        self.prefill_args_f16 = artifacts_dir + "prefill_args"
+        self.decode_args_f16 = artifacts_dir + "decode_args"
+        self.prefill_args_fp8 = artifacts_dir + "prefill_args_fp8"
+        self.decode_args_fp8 = artifacts_dir + "decode_args_fp8"
```

Comment:
Path manipulation should use https://docs.python.org/3/library/pathlib.html

These would then look like
```python
self.prefill_args_f16 = artifacts_dir / "prefill_args"
```

pathlib handles operating system path differences while direct string manipulation like you have here will not. For example, paths on Windows use `\` instead of `/`. The pathlib code treats the `/` _operator_ as an abstract path join operation (similar to https://docs.python.org/3/library/os.path.html#os.path.join) that will do the right thing when eventually converting to a concrete operating system specific path.